cmd/security-secrets-setup/README.md

```diff
@@ -2,15 +2,87 @@
+```
+
+This will create an executable located at `cmd/security-secrets-setup/` if successful.
+
+## Run **generate** subcommand
+
+To use the **generate** subcommand, go to `cmd/security-secrets-setup/` and then run
+
+```sh
+ sudo -E ./security-secrets-setup generate
+```
+
+Verify that it runs successfully with generating both Vault and Kong's TLS assets under `/run/edgex/secrets/` directory.
+
+## Run **cache** subcommand
+
+Before **cache** subcommand is issued, make sure cache dir (in this example, it is `/tmp/pki/cache`) is non-existing or empty.
+
+To use the **cache** subcommand, from the same `cmd/security-secrets-setup/`  directory run
+
+```sh
+ mkdir -p /tmp/pki/cache
+ export PKI_CACHE=/tmp/pki/cache
+ sudo -E ./security-secrets-setup cache
+```
+
+in which environment variable *PKI_CACHE* is to specify the path to the cached location.
+
+After being successfully run, the Vault and Kong TLS assets are generated and cached into the cache dir `/tmp/pki/cache` and also deployed from the cache directory into the deployed directory `/run/edgex/secrets`.  One can see the files with `ls` command for example.
+
+One can also use some file diff tool to compare the TLS files like:
+
+```sh
+ sudo diff -r /tmp/pki/cache/ca/ca.pem /run/edgex/secrets/ca/ca.pem
+```
+
+After the certificates are deployed, `security-secrets-setup` will generate a  *sentinel* file `.security-secrets-setup.complete` per certificate directory in the root deploy directory to indicate successful cache.  For example:
+
+```sh
+$ sudo diff -r /tmp/pki/cache/ /run/edgex/secrets/
+
+Only in /run/edgex/secrets/ca: .security-secrets-setup.complete
+Only in /run/edgex/secrets/edgex-kong: .security-secrets-setup.complete
+Only in /run/edgex/secrets/edgex-vault: .security-secrets-setup.complete
+```
+
+## Run **import** subcommand
+
+To use the **import** subcommand, from the same `cmd/security-secrets-setup/` directory run
+
+```sh
+ mkdir -p /tmp/pki/cache
+ export PKI_CACHE=/tmp/pki/cache
+ sudo -E ./security-secrets-setup import
+```
+
+The **import** subcommand operates differently depending on the status of cache directory:
+
+- Cache dir `/tmp/pki/cache` is empty:
+
+    In this case, import just errors out: the TLS assets are required to be in the cache directory before they can be deployed into directory `/run/edgex/secrets/`.
+
+- Cache dir `/tmp/pki/cache` has already loaded or cached with generated TLS assets before run **import** subcommand:
+
+    One should see the deploying TLS assets from the cached directory `/tmp/pki/cache` into the deployed directory `/run/edgex/secrets`.  One can verify that there are Vault and Kong TLS assets deployed from cache directory `/tmp/pki/cache` onto the directory `/run/edgex/secrets` via `ls` command.
+
+## Docker Build
+
+Use the Makefile in the root directory of the repository to build the docker container image for `security-secrets-setup`:
 
 ```sh
 make docker_security_secrets_setup
```

Comment:
When I get to the 'make docker_security_secrets_setup' step I see this error message:

```$ make docker_security_secrets_setup
docker build \
	-f cmd/security-secrets-setup/Dockerfile \
	--label "git_sha=21750bd69222753f69a7a217e480b07ca6bb6dd6" \
	-t edgexfoundry/docker-edgex-vault:21750bd69222753f69a7a217e480b07ca6bb6dd6 \
	-t edgexfoundry/docker-edgex-vault:1.1.0-dev \
	.
error checking context: 'can't stat '/home/tpm/go-jim/edgex-go/cmd/security-secrets-setup/logs/security-secrets-setup.log''.
Makefile:174: recipe for target 'docker_security_secrets_setup' failed
make: *** [docker_security_secrets_setup] Error 1
```

Looking at the directory structure of the secrets_setup I see the log file are owned by 'root':
```$ ls -la /home/tpm/go-jim/edgex-go/cmd/security-secrets-setup
-rwxrwxr-x  1 tpm  tpm      3012 Oct  2 10:28 entrypoint.sh
drwxr--r--  2 root root     4096 Oct  2 10:30 logs
-rw-rw-r--  1 tpm  tpm      4514 Oct  2 10:28 main.go
```

Reply:
yeap, please go to the `root` directory- ie. `edgex-go/` to run the docker build: `make docker_security_secrets_setup` . that way it should be fine. I'll mention that in the document about the root directory you should be run the make docker